sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -643,6 +646,87 @@ class Engine(val config: EngineConfig) {
     } yield ()
   }
 
+  /** This method checks a Dar file is self-consistent (it
+    * contains all its dependencies and only those dependencies), contains only well-formed
+    * packages (See Daml-LF spec for more details) and uses only the
+    * allowed language versions (as described by the engine
+    * config).
+    * This is not affected by [[config.packageValidation]] flag.
+    * Package in [[pkgIds]] but not in [[pkgs]] are assumed to be
+    * preloaded.
+    */
+  def validateDar(dar: Dar[(PackageId, Package)]): Either[Error.Package.Error, Unit] = {
```

Comment:
Intention is that this function will eventually replace `Engine.validatePackages`

Reply:
From an engine PoV, the Dar is either valid or not. 

Use of configuration to determine if the error is reported (by the participants package upload service) as a Grpc error or logged is made at the participant level.